hangman.py

```diff
@@ -0,0 +1,129 @@
+import random
+from words import words
+
+def hangman_pick_word():
+    """
+    Randomly picks a theme from the words.py file and converts to upper case.
+    """
+    theme = random.choice(list(words.keys()))
+    word = random.choice(words.get(theme)).upper()
+    return theme, word
+
+def hangman_generate_hint(theme, word):
+    """
+    Generates a hint based on the category randomly chosen
+    """
+    print("It\'s a {0} with {1} letters!".format(theme, len(word)))
+
+
+def hangman_print_dashes(word):
+    """
+    Generates a standard hint and the number of dashes based on the word
+    previously selected 
+    """
+    hangman_clear_screen()
+    no_of_dashes = len(word)
+    print_dashes = "_" * no_of_dashes 
+    print("Word: {0} ".format(" ".join(print_dashes)))
+    return print_dashes
+
+
+def hangman_get_letter():
+    """
+    Requests user to input a letter and converts it to upper case
+    """
+    user_guess = input("Please guess a letter: ").upper()
+    hangman_clear_screen()
+    return user_guess
+
+
+def hangman_clear_screen():
+    """
+    Clears screen based on the range specifed.
+    """
+    for i in range(1):
+        print("\n")
+
+
+def hangman_win_lose(word, guessed):
+    """
+    Checks if the player won or lost and prints the information
+    """
+    if guessed:
+        print("Congratulations! You won the game!")
+    else:
+        print("Sorry you ran out of lives! The correct answer is {0} ".format(word))
+    return 
+
+
+def hangman_algorithm(theme, word):
+    """
+    Sets user attemps and initial state of the game. Obtains user guesses and
+    checks for validity. If user guess is correct, displays the appropriate
+    letters corresponding to the guess. If the user guess is invalid, decreases
+    attempts and informs user. If user guessed a previously guessed letter, no
+    penealty for the user.
+    """
+    user_guess_list = set()
+    attempts = 9
+    guessed = False
+    valid_indices = []
+
+    print_dashes = hangman_print_dashes(word)
```

Comment:
The naming here is a bit misleading since the method does more than just printing, it also returns something.

Reply:
Actually the doc strings I added for this was misleading! It does just print dashes based on word! I've updated the doc strings!